src/Pact/Native/Keysets.hs

```diff
@@ -83,31 +81,50 @@ defineKeyset g0 fi as = case as of
     go name ks = do
       let i = _faInfo fi
 
-      (ksn, mug) <- ifExecutionFlagSet FlagDisablePact44
-        (pure (KeySetName name Nothing, Nothing))
-        (use (evalRefs . rsNamespace) >>= \case
-          Nothing ->
-            -- disallow creating keysets outside of a namespace
-            evalError' fi "Cannot define keysets outside of a namespace"
-          Just (Namespace nsn ug _ag) ->
-            case parseQualifiedKeySetName name of
-              Left {} ->
-                evalError' fi "incorrect keyset name format"
-              Right k@(KeySetName _ (Just nsn'))
-                | nsn == nsn' -> pure (k, Just ug)
-              _ -> evalError' fi "Mismatching keyset namespaces")
+      ksn <- ifExecutionFlagSet FlagDisablePact44
+        (pure $ KeySetName name Nothing)
+        (case parseAnyKeysetName name of
+           Left {} ->
+             evalError' fi "incorrect keyset name format"
+           Right ksn -> pure ksn)
 
+
+      mNs <- use $ evalRefs . rsNamespace
       old <- readRow i KeySets ksn
+
       case old of
-        Nothing -> do
-          -- enforce the user guard on first definition
-          -- and make sure we have privs
-          unlessExecutionFlagSet FlagDisablePact44 $
-            traverse_ (enforceGuard i) mug
+        Nothing -> case mNs of
+          Nothing -> do
+            unlessExecutionFlagSet FlagDisablePact44 $
+              evalError' fi "Cannot define a keyset outside of a namespace"
+
+            computeGas' g0 fi (GPreWrite (WriteKeySet ksn ks)) $
+              writeRow i Write KeySets ksn ks & success "Keyset defined"
+          Just (Namespace nsn ug _ag) -> do
+            -- enforce the user guard on first definition
+            -- and make sure we have privs
+            unlessExecutionFlagSet FlagDisablePact44 $
+              enforceGuard i ug
+
+            ksn' <- if Just nsn == _ksnNamespace ksn
```

Comment:
this behavior needs to be forked as well. Pre fork a `.` name can intro an inconsistency

Reply:
done